patches/api/0483-Change-Art-to-an-interface.patch

```diff
@@ -0,0 +1,287 @@
+-    PASSAGE(46, 4, 2),
+-    POND(47, 3, 4),
+-    SUNFLOWERS(48, 3, 3),
+-    TIDES(49, 3, 3);
++// Paper start - make interface
++public interface Art extends Keyed, org.bukkit.util.OldEnum<Art> {
++    Art KEBAB = getArt("kebab");
++    Art AZTEC = getArt("aztec");
++    Art ALBAN = getArt("alban");
++    Art AZTEC2 = getArt("aztec2");
++    Art BOMB = getArt("bomb");
++    Art PLANT = getArt("plant");
++    Art WASTELAND = getArt("wasteland");
++    Art POOL = getArt("pool");
++    Art COURBET = getArt("courbet");
++    Art SEA = getArt("sea");
++    Art SUNSET = getArt("sunset");
++    Art CREEBET = getArt("creebet");
++    Art WANDERER = getArt("wanderer");
++    Art GRAHAM = getArt("graham");
++    Art MATCH = getArt("match");
++    Art BUST = getArt("bust");
++    Art STAGE = getArt("stage");
++    Art VOID = getArt("void");
++    Art SKULL_AND_ROSES = getArt("skull_and_roses");
++    Art WITHER = getArt("wither");
++    Art FIGHTERS = getArt("fighters");
++    Art POINTER = getArt("pointer");
++    Art PIGSCENE = getArt("pigscene");
++    Art BURNING_SKULL = getArt("burning_skull");
++    Art SKELETON = getArt("skeleton");
++    Art DONKEY_KONG = getArt("donkey_kong");
++    Art EARTH = getArt("earth");
++    Art WIND = getArt("wind");
++    Art WATER = getArt("water");
++    Art FIRE = getArt("fire");
++    Art BAROQUE = getArt("baroque");
++    Art HUMBLE = getArt("humble");
++    Art MEDITATIVE = getArt("meditative");
++    Art PRAIRIE_RIDE = getArt("prairie_ride");
++    Art UNPACKED = getArt("unpacked");
++    Art BACKYARD = getArt("backyard");
++    Art BOUQUET = getArt("bouquet");
++    Art CAVEBIRD = getArt("cavebird");
++    Art CHANGING = getArt("changing");
++    Art COTAN = getArt("cotan");
++    Art ENDBOSS = getArt("endboss");
++    Art FERN = getArt("fern");
++    Art FINDING = getArt("finding");
++    Art LOWMIST = getArt("lowmist");
++    Art ORB = getArt("orb");
++    Art OWLEMONS = getArt("owlemons");
++    Art PASSAGE = getArt("passage");
++    Art POND = getArt("pond");
++    Art SUNFLOWERS = getArt("sunflowers");
++    Art TIDES = getArt("tides");
+ 
+-    private final int id, width, height;
+-    private final NamespacedKey key;
+-    private static final HashMap<String, Art> BY_NAME = Maps.newHashMap();
+-    private static final HashMap<Integer, Art> BY_ID = Maps.newHashMap();
++    private static Art getArt(final String id) {
++        return Registry.ART.get(NamespacedKey.minecraft(id));
++    }
+ 
+-    private Art(int id, int width, int height) {
+-        this.id = id;
+-        this.width = width;
+-        this.height = height;
+-        this.key = NamespacedKey.minecraft(name().toLowerCase(Locale.ROOT));
++    /**
++     * @deprecated use {@link Registry#ART}
++     */
++    @Deprecated(since = "1.21.1", forRemoval = true)
++    static @NotNull Art valueOf(final @NotNull String name) {
++        final Art art = Registry.ART.get(NamespacedKey.fromString(name.toLowerCase(Locale.ROOT)));
```

Comment:
NamespacedKey.fromString is nullable this should be checked first before blindly getting the value from the registry, same for other similar impl otherwise a NPE is thrown.